### PR TITLE
feat(core): move forEachExecutorOptions to devkit

### PR DESCRIPTION
**Pull request by Coly010**
## Current Behavior

forEachExecutorOptions lives in nrwl/workspace

## Expected Behavior

forEachExecutorOptions should live in devkit

## Related Issue(s)

Fixes #

### CODE REVIEW

**Review by jaysoo — APPROVED** [review]

**Review by leosvelperez — APPROVED** [review]

File: packages/workspace/src/utilities/executor-options-utils.ts [hunk 1]
```diff
@@ -14,6 +14,7 @@ type CallBack<T> = (
 
 /**
  * Calls a function for each different options that an executor is configured with
+ * @deprecated This function will be removed from @nrwl/workspace in version 17. Prefer importing from @nrwl/devkit.
```
Comment (AgentEnder) [inline packages/workspace/src/utilities/executor-options-utils.ts:17]: Looks like no one is using this, we can just remove it I think: https://grep.app/search?q=forEachExecutorOptions

**Review by AgentEnder — APPROVED** [review]

File: packages/workspace/src/utilities/executor-options-utils.ts (hunk 1 above)
Comment (Coly010) [reply packages/workspace/src/utilities/executor-options-utils.ts:17]: I need to start using this grep app

Comment (AgentEnder) [reply packages/workspace/src/utilities/executor-options-utils.ts:17]: Its sweet for this kind of thing, just have to take a grain of salt that its only open source projects

**Review by barbados-clemens — APPROVED** [review]

**Review by mandarini — APPROVED** [review]

File: docs/generated/devkit/nrwl_devkit.md [hunk 2]
```diff
@@ -150,6 +150,8 @@ It only uses language primitives and immutable objects
 - [detectPackageManager](../../devkit/documents/nrwl_devkit#detectpackagemanager)
 - [detectWorkspaceScope](../../devkit/documents/nrwl_devkit#detectworkspacescope)
 - [extractLayoutDirectory](../../devkit/documents/nrwl_devkit#extractlayoutdirectory)
+- [forEachExecutorOptions](../../devkit/documents/nrwl_devkit#foreachexecutoroptions)
```
Comment (FrozenPandaz) [inline docs/generated/devkit/nrwl_devkit.md:153]: I agree it should be public api... It's super useful.

But this name is ugly (my fault) and I also don't know if I like the fact that theres 2 versions...

Can we hold off on making it public api for now?

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]

**Review by xiongemi — APPROVED** [review]

**Review by FrozenPandaz — APPROVED** [review]

**Pull request merged**